types/hapi/definitions/route/route-options-response.d.ts

```diff
@@ -0,0 +1,84 @@
+import {Lifecycle, Util} from "hapi";
+import {ValidationOptions} from "joi";
+
+/**
+ * For context [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-routeoptionsresponseemptystatuscode)
+ */
+export enum EmptyStatusCode {
+    HTTP_200 = 200,
+    HTTP_204 = 204,
+}
+
+export type RouteOptionsResponseSchema = boolean | ValidationOptions | ((value: object, options: object) => any);
```

Comment:
According to the docs this should be an async function.
I think the `value` can be narrowed to `object`, `Buffer` and `string`.

Reply:
Also `options` seems to be documented rather well.

Reply:
Updated.